raid.go

```diff
@@ -91,8 +94,24 @@ func (ar RaidArrays) Measurements() MeasurementsMap {
 		results[raid.Name+".state"] = raid.State
 		results[raid.Name+".type"] = raid.Type
 
-		if raid.State == RaidStatusDegraded {
-			results[raid.Name+".failed"] = strings.Join(raid.GetFailedPhysicalDevices(), "; ")
+		failedDevices, missingCount := raid.GetFailedAndMissingPhysicalDevices()
+
+		if len(failedDevices) > 0 || missingCount > 0 {
```

Comment:
Please add a short comment what is checked here and the implications

Reply:
done